### DISCUSSION TITLE
Idea: CanvasInput for Phaser 3

### DISCUSSION THREAD

**Discussion by robinheidrich** [rexrainbow/phaser3-rex-notes#261] [Ideas]
It would be very helpful if you could use a text input in Phaser without using DOM.

There is an [example](https://codepen.io/jdnichollsc/pen/waVMdB?editors=001) how to use [CanvasInput](https://github.com/goldfire/CanvasInput) in Phaser 2, but in Phaser 3 it doesn't work anymore.

The idea: add a plugin to enable CanvasInput support in Phaser 3 (or maybe even make it a custom plugin).

_And maybe make it usable for WebGL as well._

**Comment by rexrainbow** [comment]
There has a similar one, named Hidden edit.
- [Document](https://rexrainbow.github.io/phaser3-rex-notes/docs/site/hiddeninputtext/)
- [Hidden edit with text game object](https://codepen.io/rexrainbow/pen/WNXxEMV)
- [Hidden edit with Label](https://codepen.io/rexrainbow/pen/ExbgKMe)
- [Hidden edit](https://codepen.io/rexrainbow/pen/MWOjJzZ) vs [Text edit](https://codepen.io/rexrainbow/pen/YbvwBw)

**Reply by robinheidrich** [reply]
Text edit looks interesting, just a pity that it doesn't change the mouse pointer when hovering to show that this is also an input.

I will give it a try.

**Reply by rexrainbow** [reply]
Mouse pointer changing might set at text game object. Text Edit starts when clicking text game object.
